src/core.jl

```diff
@@ -145,7 +145,11 @@ function propagate(y::Branch, rvs_tape::Tape)
     kwargs = getfield(y, :kwargs)
     xs = map(unbox, args)
     xids = map(pos, args)
+
+    ȳ isa AbstractZero && return nothing
```

Comment:
We should think about introducing a function barrier here.
But I think profiling around that is out of scope for this.
but maybe we can move this to right after `ȳ` first is unpacked?